src/go/metadata/metadata_fetcher_test.go

```diff
@@ -309,9 +311,15 @@ func TestFetchGCPAttributes(t *testing.T) {
 
 		mf := NewMockMetadataFetcher(mockBaseUrl, time.Now())
 
-		attrs, err := mf.FetchGCPAttributes()
-		if err != nil && tc.expectedGCPAttributes != nil {
-			t.Errorf(errorTmpl, tc.desc, nil, err)
+		attrs, actualError := mf.FetchGCPAttributes()
+		// if expected error is set, actual error should be equal to the expected error
+		if tc.expectedError != "" && actualError.Error() != tc.expectedError {
+			t.Errorf(errorTmpl, tc.desc, tc.expectedError, actualError)
```

Comment:
The error should be `Test: %s\n ExpectedError: %v\n ActualError: %v"`

Reply:
Line 302 defines `errorTmpl := "Test: %s\n  Expected: %v\n  Actual: %v"`, which is a general statement that already covers the error case.

Reply:
It isn't cleared about the mismatch is for the expected resp or the expected error

Reply:
Sure, added `attrsTmpl` to distinguish the attrs and error templates. Thanks.